### PR TITLE
bugfix: create container with invalid name

### PR DESCRIPTION
**Pull request by lang710**
Signed-off-by: Lang Chi <21860405@zju.edu.cn>

### Ⅰ. Describe what this PR did
bugfix: create contaienr with invalid name

### Ⅱ. Does this pull request fix one issue?

#2918 

### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)
Added.

### Ⅳ. Describe how to verify it
root@compatibility:~/go/src/github.com/alibaba/pouch# pouch create --name new:invalid busybox
Error: failed to create container: {"message":"Invalid container name (new:invalid), only [a-zA-Z0-9][a-zA-Z0-9_.-] are allowed"}

### CODE REVIEW

**Comment by ZYecho** [comment]
LGTM

File: test/cli_create_test.go [hunk 1]
```diff
@@ -608,3 +608,12 @@ func (suite *PouchCreateSuite) TestCreateWithoutNvidiaConfig(c *check.C) {
 	}
 	c.Assert(result[0].HostConfig.Resources.NvidiaConfig, check.IsNil)
 }
+
+// TestCreateWithInvalidName tests creating container with invalid name.
+func (suite *PouchRunSuite) TestCreateWithInvalidName(c *check.C) {
+	name := "new:invalid"
+	res := command.PouchRun("create", "--name", name, busyboxImage)
+	if !strings.Contains(res.Stdout(), "Invalid container name") {
```
Comment (zhuangqh) [inline test/cli_create_test.go]: clean the container `defer DelContainerForceMultyTime(c, name)`

Comment (lang710) [reply test/cli_create_test.go]: OK!

File: test/cli_create_log_options_test.go [hunk 2]
```diff
@@ -78,7 +78,7 @@ func (suite *PouchCreateLogOptionsSuite) TestOK(c *check.C) {
 			driver:   "json-file",
 			expected: nil,
 		}, {
-			cname:   "TestCreateLogOptions_jsonfile_tag=1",
+			cname:   "TestCreateLogOptions_jsonfile_tag_1",
```
Comment (ZYecho) [inline test/cli_create_log_options_test.go]: why change that?

Comment (ZYecho) [reply test/cli_create_log_options_test.go]: seems it verify condition that logOpts: []string{"tag=1"}?

Comment (lang710) [reply test/cli_create_log_options_test.go]: Creating a container name with an equal sign is illegal.

**Comment by ZYecho** [comment]
LGTM

**Pull request merged**